crates/libtiny_tui/src/msg_area/mod.rs

```diff
@@ -128,12 +133,13 @@ impl MsgArea {
 // Scrolling
 
 impl MsgArea {
-    fn lines_height(&mut self) -> i32 {
+    /// The total number of visible lines if each Line was rendered at the current screen width
+    fn total_visible_lines(&mut self) -> i32 {
         match self.lines_height {
             Some(height) => height,
             None => {
                 let mut total_height = 0;
-                for line in &mut self.lines {
+                for line in self.lines.iter_mut() {
```

Comment:
Why change this?

I'm a bit surprised that we update all lines (or "messages" as I suggest above) here. If I'm currently showing lines/messages  in range `n..m` (in Rust range syntax), and make the messaging area larger, most of the time I only need to know the a few lines before `n` and/or after `m` (depending on the implementation). Similarly when making it smaller, I'll need to update a smaller range in `n..m` (e.g. `n+x..m-y` where `x, y >= 0`).

So it's wasteful to update all the lines here. I think most of the values computed won't even be used most of the time.

Not really related to this PR though, I will open an issue about this.

Reply:
Yes, there can definitely be a more efficient way to handle resizing/scrolling.